core/watcher/notify.go

```diff
@@ -142,3 +161,7 @@ func (nw *NotifyWorker) Report() map[string]interface{} {
 	}
 	return nil
 }
+
+func (nw *NotifyWorker) scopedContext() (context.Context, context.CancelFunc) {
```

Comment:
I recall there was some dissent around my `BaseWatcher` that included the `Kill` and `Wait` commands, so that embedding it reduced boilerplate in embedders, but I like that style of declarative programming - "putting" the behaviour.

Is it worth doing something like that here, so that we can embed scoped context behaviour into all catacomb-based workers?

Reply:
Yeah, maybe there should be a base/common catacomb worker inside of Juju that gives you these sort of basic functionality.